### PR TITLE
chore: commit grammar.json in tree-sitter-wing

### LINKED ISSUE #6043: Tree-sitter: consider committing `grammar.json`
### Use Case

Install the grammar via `nvim-treesitter` (1.0, in development).

### Proposed Solution

While keeping `parser.c` out of version control is reasonable (and in fact recommended by upstream), the generated `grammar.json` is much smaller and has meaningful diffs. The benefit of keeping this in the repo is that downstream users (like nvim-treesitter) do not require `node` for building and can just do `tree-sitter g grammar.json && tree-sitter build -o parser.so` (which is what nvim-treesitter is transitioning to for 1.0).

### Implementation Notes

In addition, consider making proper releases including the generated artifacts (which is what nvim-treesitter will require for "stable" parsers). Upstream is in the process of adding reusable workflows and `tree-sitter` support to make this easier.

### Component

tree-sitter-wing

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.
* If this issue is labeled **needs-discussion**, it means the spec has not been finalized yet. Please reach out on the #dev channel in the [Wing Slack](https://t.winglang.io/slack).

### PR DESCRIPTION
**Pull request by MarkMcCulloh**
Fixes #6043

Grammar updates aren't happening too frequently, so this hopefully shouldn't cause much trouble.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/tree-sitter-wing/.gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+/src/grammar.json linguist-generated
```
Comment (staycoolcall911) [inline libs/tree-sitter-wing/.gitattributes:1]: I added the pr/e2e-full label on this PR, but maybe it shouldn't matter? If that's not needed, you can remove it before you merge that in.

**Review by staycoolcall911 — APPROVED** [review]

File: libs/tree-sitter-wing/.gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /src/*
 !/src/scanner.c
+!/src/grammar.json
```
Comment (eladb) [inline libs/tree-sitter-wing/.gitignore:3]: will this get automatically updated by the self mutation in case it's changed?

**Review by eladb — APPROVED** [review]

File: libs/tree-sitter-wing/.gitignore (hunk 2 above)
Comment (MarkMcCulloh) [reply libs/tree-sitter-wing/.gitignore:3]: Yep it should

File: libs/tree-sitter-wing/.gitattributes (hunk 1 above)
Comment (MarkMcCulloh) [reply libs/tree-sitter-wing/.gitattributes:1]: Shouldn't be needed, the file already exists in the built version

**Pull request merged**